cockatrice/src/dialogs/dlg_settings.cpp

```diff
@@ -588,8 +588,8 @@ DeckEditorSettingsPage::DeckEditorSettingsPage()
     connect(urlList->model(), SIGNAL(rowsMoved(const QModelIndex, int, int, const QModelIndex, int)), this,
             SLOT(urlListChanged(const QModelIndex, int, int, const QModelIndex, int)));
 
-    for (int i = 0; i < SettingsCache::instance().downloads().getCount(); i++)
-        urlList->addItem(SettingsCache::instance().downloads().getDownloadUrlAt(i));
+    foreach (QString url, SettingsCache::instance().downloads().getAllURLs())
```

Comment:
Please don't use the Qt foreach, but instead the CPP for (auto i : list) approach